src/Listener.cpp

```diff
@@ -143,9 +142,38 @@ std::string Listener::ThreadID()  {
 bool Listener::NotificationReceived() {
    std::vector<std::string> dataFromQueue;
    mQueueReader->GetShot(getShotTimeout, dataFromQueue);
-   return MessageHasPayload(dataFromQueue);
+   bool notificationReceived = MessageHasPayload(dataFromQueue);
+   ClearMessages();
```

Comment:
Oh you mean that it needs to go inside the ```if()``` statement. I thought you meant after the if block. You're right I'll change that